### PR TITLE
Added Github action to automatically deploy to PyPi

### LINKED ISSUE #3: Set up GitHub Actions to build and release to PyPi
This should be our default CI/CD pipeline for open source projects (which this will be!).

It is not immediately obvious to me if we should remove `build_release.sh`, decrease its scope, or just call it from the Action.

Secrets are managed differently (better), so we can remove the check for `pypirc`, and just pass `TWINE_USERNAME` and `TWINE_PASSWORD` as ENV vars from SECRETS.

### PR DESCRIPTION
**Pull request by lucfons**
closes #3

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Publish to testpypi repo
+
+on: push
+
+jobs:
+  build-n-publish:
+    name: Build and publish Python 🐍 distributions 📦 to TestPyPI
```
Comment (sam-qordoba) [inline .github/workflows/main.yml:7]: Does this publish to TestPyPi, or PyPi?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@master
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+    - name: Run build script
+      run: >-
+        bash build_release.sh
```
Comment (sam-qordoba) [inline .github/workflows/main.yml:17]: If the tests fail here, does it prevent it from publishing (that is the desired behavior)

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    - name: Publish distribution 📦 to PyPI
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        password: ${{ secrets.pypi_password }}
```
Comment (sam-qordoba) [inline .github/workflows/main.yml:21]: I assume we have to configure this in the main repo before merging this?

Comment (lucfons) [reply .github/workflows/main.yml:21]: I have already configured this for the main repo.

File: .github/workflows/main.yml (hunk 1 above)
Comment (lucfons) [reply .github/workflows/main.yml:7]: It publishes to PyPi, I had tested it using TestPyPI, then changed it so it publishes to PyPi but forgot to change the name of the action

Comment (lucfons) [reply .github/workflows/main.yml:7]: I will do so now

File: .github/workflows/main.yml (hunk 3 above)
Comment (sam-qordoba) [reply .github/workflows/main.yml:21]: And the secret is a generated API token? does it expire?

**Comment by sam-qordoba** [comment]
@lucfons - about the merge conflicts. Have you ever resolved those? It shouldn't be too bad... they're caused because `Qordobacode:master` has changes that `lucfons:master` doesn't have. Sometimes git can figure out what wins, but not here. Do you think you can resolve them?

**Pull request closed without merging**